hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/cli/GenericCli.java

```diff
@@ -44,13 +45,18 @@ public class GenericCli implements Callable<Void>, GenericParentCommand {
   private boolean verbose;
 
   @Option(names = {"-D", "--set"})
-  private Map<String, String> configurationOverrides = new HashMap<>();
+  public void setConfigurationOverrides(Map<String, String> configOverrides) {
+    configOverrides.forEach(getOzoneConf()::set);
+  }
 
   @Option(names = {"-conf"})
-  private String configurationPath;
+  public void setConfigurationPath(String configPath) {
+    getOzoneConf().addResource(new Path(configPath));
+  }
 
+  private final Supplier<OzoneConfiguration> configSupplier =
+      MemoizedSupplier.valueOf(OzoneConfiguration::new);
   private final CommandLine cmd;
-  private OzoneConfiguration conf;
```

Comment:
Since almost all commands use the config, I think we can create the config eagerly:

```java
  private final OzoneConfiguration conf = new OzoneConfiguration();
```

Reply:
Makes sense. Will update the PR.